pkg/signing/signing.go

```diff
@@ -322,23 +324,27 @@ func Verify(cpath, keyServiceURI string, id uint32, isGroup bool, authToken stri
 		if err != nil {
 			// if theres a error, thats proboly becuse we dont have a local key
 
-			// download the key
-			sylog.Infof("Downloading key: %s...", fingerprint[24:])
-			netlist, err := sypgp.FetchPubkey(fingerprint, keyServiceURI, authToken, noPrompt)
-			if err != nil {
-				return fmt.Errorf("could not fetch public key from server: %s", err)
-			}
-			sylog.Verbosef("key retrieved successfully!")
-
-			block, _ := clearsign.Decode(data)
-			if block == nil {
-				return fmt.Errorf("failed to parse signature block")
-			}
-
-			// verify the container
-			signer, err = openpgp.CheckDetachedSignature(netlist, bytes.NewBuffer(block.Bytes), block.ArmoredSignature.Body)
-			if err != nil {
-				return fmt.Errorf("signature verification failed: %s", err)
+			if !localVerify {
+				// download the key
+				sylog.Infof("Downloading key: %s; not in keyring of trust.", fingerprint[24:])
```

Comment:
What happens here if there are multiple signature blocks? Is the key downloaded multiple times?

Reply:
Yes, all the keys will be downloaded, like:

```bash
$ singularity verify test.sif 
WARNING: sylabs-token files are deprecated. Use 'singularity remote' to manage remote endpoints and tokens.
Verifying image: test.sif
INFO:    Downloading key: F1E47D55A7F3A56C; not in keyring of trust.
INFO:    Downloading key: 25C5D19789677B67; not in keyring of trust.
Data integrity checked, authentic and signed by:
	westleyk (examples) <westley@sylabs.io>, Fingerprint 4E28E95609D65D3C5BEA9731F1E47D55A7F3A56C
	Sylabs Admin <support@sylabs.io>, Fingerprint 8883491F4268F173C6E5DC49EDECE4F3F38D871E
	Westley (test) <westley@sylabs.io>, Fingerprint 93EE8637AF8AF616AD54EFFF25C5D19789677B67
```

Reply:
This is fine, I was thinking multiple signature blocks with the same signature